core/libraries/rest_api/RestIncomingQueryParamMetadata.php

```diff
@@ -0,0 +1,705 @@
+
+    /**
+     * If query_param_value were in the simplified specific operator structure, change it into the legacy structure.
+     * @since $VID:$
+     * @throws RestException
+     */
+    private function transformSimplifiedSpecifiedOperatorSyntaxIntoStandardSyntax()
+    {
+        if ($this->valueIsAssociativeArray()) {
+            $this->assertSimplifiedSpecifiedOperator();
+            $query_param_value = $this->getQueryParamValue();
+            $sub_array_value = reset($query_param_value);
+            $sub_array_key = key($query_param_value);
+            $this->assertSubValueIsntArray($sub_array_key, $sub_array_value);
+            // they're doing something like "&where[EVT_ID][IN]=1,2,3" or "&where[EVT_ID][>]=5"
+            if ($this->subArrayKeyIsNonBinaryOperator($sub_array_key)) {
+                $this->setQueryParamValue(array(
+                    $sub_array_key,
+                    $this->extractQuickStyleSpecifiedOperatorValue($sub_array_value)
+                ));
+            } elseif ($this->subArrayKeyIsUnaryOperator($sub_array_key)) {
+                $this->setQueryParamValue(array($sub_array_key));
+            } else {
+                $this->setQueryParamValue(array($sub_array_key, $sub_array_value));
+            }
+        }
+    }
+
+    /**
+     * Returns true is the value is an array using the legacy structure to specify the operator. Eg `array('!=',123)`.
+     * @since $VID:$
+     * @return bool
+     */
+    private function valueIsLegacySpecifiedOperator()
+    {
+        $valid_operators = $this->getContext()->getModel()->valid_operators();
+        $query_param_value = $this->getQueryParamValue();
+        return isset($query_param_value[0])
+            && isset($valid_operators[ $query_param_value[0] ]);
+    }
+
+    /**
+     * Returns true if the value specified operator accepts arbitrary number of arguments, like "IN".
+     * @since $VID:$
+     * @param $operator
+     * @return bool
+     */
+    private function operatorIsNAry($operator)
+    {
+        $valueArray = $this->getQueryParamValue();
+        return array_key_exists(
+            $operator,
+            $this->getContext()->getModel()->valid_in_style_operators()
+        )
+            && isset($valueArray[1])
+            && is_array($valueArray[1])
+            && !isset($valueArray[2]);
+    }
+
+    /**
+     * Returns true if the operator accepts 3 arguments (eg "BETWEEN").
+     * @since $VID:$
+     * @param $operator
+     */
+    private function operatorIsTernary($operator)
+    {
+        $query_param_value = $this->getQueryParamValue();
+        return array_key_exists($operator, $this->getContext()->getModel()->valid_between_style_operators())
+            && isset($query_param_value[1])
+            && is_array($query_param_value[1])
+            && isset($query_param_value[1][0], $query_param_value[1][1])
+            && !isset($query_param_value[1][2])
+            && !isset($query_param_value[2]);
+    }
```

Comment:
great compromise! 
I can go a touch overboard with trying to think in terms of objects for everything,
but this more than sufficient for making the logic that is now in `determineModelValueGivenRestInputArray()` very readable.
No one will is going to get overwhelmed looking at any of this now because the big wall of nondescript code has been broken apart into these easily understood chunks.
good stuff!

Reply:
thanks, ya although we can keep improving these private methods down the road too